concrete/src/Job/JobFactory.php

```diff
@@ -0,0 +1,278 @@
+    {
+        $jcl = $this->getJobClassLocations();
+        $pkgHandle = null;
+
+        //check for the job file in the various locations
+        $db = $this->app['database']->connection();
+        $pkgID = $db->fetchColumn("SELECT pkgID FROM Jobs WHERE jHandle = ?", [
+            $jHandle,
+        ]);
+
+        if ($pkgID > 0) {
+            $pkgHandle = PackageList::getHandle($pkgID);
+            if ($pkgHandle) {
+                $jcl[] = DIR_PACKAGES.'/'.$pkgHandle.'/'.DIRNAME_JOBS;
+                $jcl[] = DIR_PACKAGES_CORE.'/'.$pkgHandle.'/'.DIRNAME_JOBS;
+            }
+        }
+
+        foreach ($jcl as $jobClassLocation) {
+            //load the file & class, then run the job
+            $path = $jobClassLocation.'/'.$jHandle.'.php';
+            if (file_exists($path)) {
+                $className = $this->getClassName($jHandle, $pkgHandle);
+                $j = $this->app->make($className);
+                $j->jHandle = $jHandle;
+                if (count($jobData)) {
+                    $j->setPropertiesFromArray($jobData);
+                }
+
+                return $j;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param bool $includeConcreteDirJobs
+     *
+     * @return array
+     */
+    protected function getJobClassLocations($includeConcreteDirJobs = true)
+    {
+        if (!$includeConcreteDirJobs) {
+            $jobClassLocations = [DIR_FILES_JOBS];
+        } else {
+            $jobClassLocations = [DIR_FILES_JOBS, DIR_FILES_JOBS_CORE];
+        }
+
+        return $jobClassLocations;
+    }
+
+    /**
+     * @param string      $jHandle
+     * @param null|string $pkgHandle
+     *
+     * @return string
+     */
+    protected function getClassName($jHandle, $pkgHandle = null)
+    {
+        $class = overrideable_core_class('Job\\'.camelcase($jHandle), DIRNAME_JOBS.'/'.$jHandle.'.php', $pkgHandle);
+
+        return $class;
+    }
+
+    /**
+     * Return list of Jobs from a specific package.
+     *
+     * @param object $pkg
+     *
+     * @return Job[]
+     */
+    public function fromPackage($pkg)
+    {
+        $list = [];
+
+        $db = $this->app['database']->connection();
+        $r = $db->executeQuery("ELECT jHandle FROM Jobs WHERE pkgID = ? ORDER BY jHandle ASC", [
+            $pkg->getPackageID(),
+        ]);
+
+        while ($row = $r->fetchAssoc()) {
+            $list[] = $this->getJobObjByHandle($row['jHandle']);
+        }
+
+        return $list;
+    }
+
+    /**
+     * @deprecated Use 'fromPackage($pkg)'
+     *
+     * @param $pkg
+     *
+     * @return array
+     */
+    public function getListByPackage($pkg)
+    {
+        return $this->fromPackage($pkg);
+    }
+
+    /**
+     * @deprecated Use 'installed' and 'scheduled' methods instead.
+     *
+     * @param bool $scheduledOnly
+     *
+     * @return Job[]
+     */
+    public function getList($scheduledOnly = false)
```

Comment:
I'm confused by these deprecated methods in this class. Since this is a new class we shouldn't need to add new methods then immediately deprecate them other than for things like that static application setter. `Package::getList()` should just call the non-deprecated method instead.